topics/go/language/embedding/exercises/exercise1/exercise1.go

```diff
@@ -1,64 +1,110 @@
 // All material is licensed under the Apache License Version 2.0, January 2004
 // http://www.apache.org/licenses/LICENSE-2.0
 
-// Copy the code from the template. Declare a new type called hockey
-// which embeds the sports type. Implement the finder interface for hockey.
-// When implementing the find method for hockey, call into the find method
-// for the embedded sport type to check the embedded fields first. Then create
-// two hockey values inside the slice of finders and perform the search.
+// Sample program to show how you can use embedding to reuse behavior from
+// another type and override specific methods.
 package main
 
 import (
 	"fmt"
-	"strings"
+	"log"
+	"time"
 )
 
-// finder defines the behavior required for performing matching.
-type finder interface {
-	find(needle string) bool
+// Document is the core data model we are working with.
+type Document struct {
+	Key   string
+	Title string
 }
 
-// sport represents a sports team.
-type sport struct {
-	team string
-	city string
-}
+// ==================================================
+
+// Feed is a type that knows how to fetch Documents.
+type Feed struct{}
 
-// find checks the value for the specified term.
-func (s *sport) find(needle string) bool {
-	return strings.Contains(s.team, needle) || strings.Contains(s.city, needle)
+// Count tells how many documents are in the feed.
+func (f *Feed) Count() int {
+	return 42
 }
 
-// hockey represents specific hockey information.
-type hockey struct {
-	sport
-	country string
+// Fetch simulates looking up the document specified by key. It is slow.
+func (f *Feed) Fetch(key string) (Document, error) {
+
+	time.Sleep(1000 * time.Millisecond)
+
+	doc := Document{
+		Key:   key,
+		Title: "Title for " + key,
+	}
+	return doc, nil
 }
 
-// find checks the value for the specified term.
-func (h *hockey) find(needle string) bool {
-	return h.sport.find(needle) || strings.Contains(h.country, needle)
+// ==================================================
+
+// CachingFeed keeps a local copy of Documents that have already been
+// retrieved. It embeds Feed to get the Fetch and Count behavior but
+// "overrides" Fetch to have its cache.
+type CachingFeed struct {
+	docs map[string]Document
+	*Feed
 }
 
-func main() {
+// NewCachingFeed initalizes a CachingFeed for use.
+func NewCachingFeed(f *Feed) *CachingFeed {
+	fc := CachingFeed{
```

Comment:
Put the return on the construction since this is the only value being returned.

Reply:
I updated my comment because I realized Github's code formatting is slightly different than what I am used to in Slack. The first line of each example was being hidden.

Reply:
Put it on the return

Reply:
You don't need a variable in this case.